service/storage_service.cc

```diff
@@ -3100,15 +3100,20 @@ storage_service::stream_ranges(std::unordered_map<sstring, std::unordered_multim
             dht::token_range r = end_point_entry.first;
             inet_address endpoint = end_point_entry.second;
             ranges_per_endpoint[endpoint].emplace_back(r);
+            co_await coroutine::maybe_yield();
         }
-        streamer->add_tx_ranges(keyspace, std::move(ranges_per_endpoint));
+        streamer.add_tx_ranges(keyspace, std::move(ranges_per_endpoint));
     }
-    return streamer->stream_async().then([streamer] {
+    std::exception_ptr ep;
+    try {
+        co_await streamer.stream_async();
         slogger.info("stream_ranges successful");
-    }).handle_exception([] (auto ep) {
+        co_return;
+    } catch (...) {
+        ep = std::current_exception();
         slogger.warn("stream_ranges failed: {}", ep);
-        return make_exception_future<>(std::move(ep));
-    });
+    }
+    co_await coroutine::return_exception_ptr(std::move(ep));
```

Comment:
Why do we need this coroutine::return_exception_ptr? Can't we just throw in the catch branch above?

Reply:
we can, but rethrowing is less efficient and
the coroutine boilerplate code will need to catch it and convert to an exceptional future as done here.

Reply:
But the code would be simpler and less home grown apis. "Less efficient" is less important here because the code is on the control plane and is rare.

Reply:
no problem, will change